template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/atlas/AtlasKnoxRoleConfigProvider.java

```diff
@@ -2,30 +2,68 @@
 
 import static com.sequenceiq.cloudbreak.cmtemplate.configproviders.ConfigUtils.config;
 
+import java.util.ArrayList;
+import java.util.Comparator;
 import java.util.List;
 import java.util.Objects;
 
-import javax.inject.Inject;
-
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
 import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.common.StackType;
 import com.sequenceiq.cloudbreak.api.service.ExposedServiceCollector;
-import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
 import com.sequenceiq.cloudbreak.cmtemplate.configproviders.AbstractRoleConfigProvider;
+import com.sequenceiq.cloudbreak.common.type.Versioned;
 import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+import com.sequenceiq.cloudbreak.util.VersionComparator;
 
 @Component
 public class AtlasKnoxRoleConfigProvider extends AbstractRoleConfigProvider {
 
+    public static final String MIN_VERSION_FOR_DIFFERENCIAL_BACKUP = "7.2.7";
+
     private static final String ATLAS_KNOX_CONFIG = "atlas_authentication_method_trustedproxy";
 
-    @Inject
-    private ExposedServiceCollector exposedServiceCollector;
+    private static final String ATLAS_DIFFERENTIAL_AUDIT_CONFIG = "atlas.entity.audit.differential";
+
+    private static final Logger LOG = LoggerFactory.getLogger(AtlasKnoxRoleConfigProvider.class);
+
+    private final ExposedServiceCollector exposedServiceCollector;
+
+    private final EntitlementService entitlementService;
+
+    private final ObjectMapper objectMapper;
+
+    public AtlasKnoxRoleConfigProvider(EntitlementService entitlementService,
+            ExposedServiceCollector exposedServiceCollector,
+            ObjectMapper objectMapper) {
+        this.entitlementService = entitlementService;
+        this.exposedServiceCollector = exposedServiceCollector;
+        this.objectMapper = objectMapper;
+    }
 
     @Override
     public List<ApiClusterTemplateConfig> getRoleConfigs(String roleType, TemplatePreparationObject source) {
-        return List.of(config(ATLAS_KNOX_CONFIG, "true"));
+        List<ApiClusterTemplateConfig> configs = new ArrayList<>();
+        if (isSDXOptimizationEnabled(source)) {
+            LOG.info(String.format("SDX Optimization Enabled for %s.", roleType));
+            LOG.debug("Atlas source: {}", source);
```

Comment:
This line is needed? What's the output, I see no proper toString for `TemplatePreparationObject`